File: gaia-sdk-javascript/src/graphql/request/input/CreateRoleImpulse.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * The specification to create a role instance
 */
export class CreateRoleImpulse {
public _typeName = "CreateRoleImpulse";
    private tenantId:Uuid;
    private name:String;
    private permissions:Array<String>;

    constructor (tenantId:Uuid, name:String, permissions:Array<String>) {
        this.tenantId = tenantId;
        this.name = name;
        this.permissions = permissions;
    }
}

File: gaia-sdk-python/gaia_sdk/graphql/request/input/CreateRoleImpulse.py


from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class CreateRoleImpulse():
    """
    The specification to create a role instance
    """
    tenantId: str
    name: str
    permissions: List[str]

    def __init__(self, tenantId: str, name: str, permissions: List[str]):
        self.tenantId = tenantId
        self.name = name
        self.permissions = permissions

    def __eq__(self, other):
        if type(other) is type(self):
            return self.tenantId == other.tenantId and self.name == other.name and self.permissions == other.permissions
        return False

    def __repr__(self):
        return {'tenantId': self.tenantId, 'name': self.name, 'permissions': self.permissions}
